core/src/main/java/org/infinispan/interceptors/impl/CacheMgmtInterceptor.java

```diff
@@ -400,6 +404,52 @@ public long getOffHeapMemoryUsed() {
       return allocator.getAllocatedAmount();
    }
 
+   @ManagedAttribute(
+         description = "Amount of nodes required to guarantee data consistency",
+         displayName = "Required Minimum Nodes",
+         displayType = DisplayType.SUMMARY
+   )
+   public int getRequiredMinimumNumberOfNodes() {
+      return calculateRequiredMinimumNumberOfNodes(cache);
+   }
+
+   public static int calculateRequiredMinimumNumberOfNodes(AdvancedCache<?, ?> cache) {
+      Configuration config = cache.getCacheConfiguration();
+
+      ClusteringConfiguration clusteringConfiguration = config.clustering();
+      CacheMode mode = clusteringConfiguration.cacheMode();
+      if (mode.isReplicated() || !mode.isClustered()) {
+         // Local and replicated only require the 1 node to keep the data
+         return 1;
+      }
+      CacheTopology cacheTopology = cache.getDistributionManager().getCacheTopology();
+      if (mode.isInvalidation()) {
+         // Invalidation requires all as we don't know what data is installed on which
+         return cacheTopology.getMembers().size();
+      }
+      int numMembers = cacheTopology.getMembers().size();
+      int numOwners = clusteringConfiguration.hash().numOwners();
+      int minNodes;
+      if (mode.isScattered()) {
+         minNodes = numMembers - 1;
```

Comment:
Why are you special-casing scattered mode, which is a dist mode with 2 owners and would therefore be satisfied by the DIST branch below.

Reply:
Scattered afaik doesn't use num owners afaik, so someone could put any value they wanted. So I just hard coded this to assume 2 in that case.

Reply:
Should we say 0? Also what about local then as well? In either case it could be in a state where it would say "don't ever shut down a node".